### ISSUE TITLE
Check article and code matches

### ISSUE THREAD

**Issue by richelbilderbeek** [richelbilderbeek/razzo#244]
I need this before simplifying ?`create_razzo_paramses_file`.

Start has been made:

```r
test_that("matches article", {
  pir_params <- create_razzo_pir_params(
    has_candidates = TRUE,
    has_twinning = TRUE
  )

  ##############################################################################
  # pir_params$twinning_params
  ##############################################################################
  expect_true(!beautier::is_one_na(pir_params$twinning_params))
  expect_equal(pir_params$twinning_params$twin_model, "birth_death")
  expect_equal(pir_params$twinning_params$method, "random_tree")
  ##############################################################################
  # pir_params$alignment_params
  ##############################################################################
  expect_equal(
    pirouette::create_blocked_dna(length = 1000),
    pir_params$alignment_params$root_sequence
  )
  expect_equal(
    pirouette::create_standard_mutation_rate,
    pir_params$alignment_params$mutation_rate
  )
  expect_equal(
    pir_params$alignment_params$site_model$name,
    "JC69"
  )
  expect_equal(
    pir_params$alignment_params$clock_model$name,
    "strict"
  )
  # Etcetera ...
})
```

**Comment by Giappo** [comment]
This is what we wrote in the article:
- [ ] **mbd params:**
- [ ] lambda = c(0.2)
- [ ] mu = c(0, 0.15)
- [ ] nu = c(1.0, 1.5, 2.0, 2.5)
- [ ] q = c(0.1, 0.15, 0.2)
- [ ] crown_age = 6;
- [ ] **alignment_model**:
- [ ] sequence_length = 1000;
- [ ] site_model = "JC69";
- [ ] clock_model = "strict";
- [ ] mutation rate = 0.5 * crown_age = 3;
- [ ] **inference_model:**
- [ ] site_model = "JC69";
- [ ] clock_model = "strict";
- [ ] tree_prior = "BD"
- [ ] MRCA prior = N(mu = crown_age, sigma = 0.01);
- [ ] chain_length = 1e7;
- [ ] sample_every = 1e3;
- [ ] **error_measure_params:**
- [ ] error_function = nltt;
- [ ] burn_in = 0.1
- [ ] **twin:**
- [ ] tree_prior = "BD";
- [ ] **experiments**
- [ ] generative = c(tree_prior = "BD", site_model = "JC69", clock_model = "strict");
- [ ] candidates = 39 candidate models, which are the results of all possible combinations of 5 tree priors, four site models and two clock models (minus the generative one);

Question: why the mutation rate is only half of the crown age? From previous conversations with you I understood it would be better to have mutation_rate = 1 * crown_age. Am I wrong?

**Comment by Giappo** [comment]
Now running the call:
`pir_params <- create_razzo_pir_params(has_candidates = TRUE, has_twinning = TRUE)`
on the cluster and comparing the results with the article

**Comment by Giappo** [comment]
This call is not enough.
Can this do the job?
`razzo_paramses <- create_razzo_paramses(project_folder_name = peregrine::get_pff_tempdir(),mbd_paramses =create_mbd_paramses(n_replicates = 1))`

**Comment by Giappo** [comment]
I suggest to create a single test script to check that all the possible parameters are in accordance with the manuscript.
We can do it calling the function that generates all the parameters for the experiment (which I suppose is the one reported in the comment above) and then check all the arguments. I think that dividing all in many scripts makes everything much harder.
@richelbilderbeek do you agree? Is the call that I reported the best one to use?

**Comment by richelbilderbeek** [comment]
> Is the call that I reported the best one to use?

I definitely agree that `create_razzo_paramses` is the function to test.

> I suggest to create a single test script to check that all the possible parameters are in accordance with the manuscript. I think that dividing all in many scripts makes everything much harder.

I don't think duplicating code (by putting it in another script) is the superior option. I am definitely aware that the checks are done on a per-component basis, yet I also think this is superior over one huge function. If each setting is tested in at least one place, that suffices. What I want to have checked is: 'Did we miss something?'

What do you think: did we miss something?

**Comment by richelbilderbeek** [comment]
> Question: why the mutation rate is only half of the crown age? From previous conversations with you I understood it would be better to have mutation_rate = 1 * crown_age. Am I wrong?

You are right for the ideal mutation scenario in which each nucleotide mutates exactly once. This scenario break if by chance there are more mutations that the number of nucleotides. For that reason there is an arbitrarily-sized extra chunk of nucleotides.

(it is a hard question to answer 'What is the mutation rate that gives an alignment with the most information?' I have looked in to this, e.g. [edible](https://github.com/richelbilderbeek/edible) tries to answer it, but in the end I could not find a clear answer)

**Comment by richelbilderbeek** [comment]
> This call is not enough. [...]

To run it on the cluster, you may like ['Cluster usage' on the razzo_project page](https://github.com/richelbilderbeek/razzo_project#cluster-usage) :+1:

**Comment by richelbilderbeek** [comment]
> What do you think: did we miss something?

@Giappo: one option to assure this, is to use functions like `get_razzo_crown_age` that is the only place where the crown age is hard-coded. If that value matches the article, then we know life is great :rainbow:
